rest_framework_api_key/permissions.py

```diff
@@ -7,12 +8,11 @@
 
 
 class HasAPIKey(permissions.BasePermission):
-    def has_permission(self, request, view) -> bool:
+    def has_permission(self, request, view) -> tuple:
         key = _get_key(request)
 
         if not key:
-            return False
-
+            return False, None
         return APIKey.objects.is_valid(key)
```

Comment:
I only tested on `EndpointMethodPermission | permissions.IsAuthenticated` but yes doesn't look like this will work on `HasAPIKey`.

The reason for doing this was to pick up the API key object in the db, but I can use the prefix as this is unique across API keys? e.g. https://github.com/florimondmanca/djangorestframework-api-key/blob/master/rest_framework_api_key/models.py#L30